src/cards/target_ricochet_arrow.ts

```diff
@@ -0,0 +1,127 @@
+    ignoreRange: true,
+    // so that you can fire the arrow at targets out of range
+    allowNonUnitTarget: true,
+    // This ensures that "target scamming" doesn't work with target arrow
+    // due to it being able to fire out of range
+    noInitialTarget: true,
+    requiresFollowingCard: true,
+    animationPath: '',
+    sfx: '',
+    description: 'spell_target_ricochet_arrow',
+    effect: async (state, card, quantity, underworld, prediction) => {
+      // - - - - - Start copied from arrow.ts - - - - -
+      let targets: Vec2[] = getCurrentTargets(state);
+      targets = targets.length ? targets : [state.castLocation];
+      let timeoutToNextArrow = 200;
+      for (let target of targets) {
+        let casterPositionAtTimeOfCast = state.casterPositionAtTimeOfCast;
+
+        const startPoint = casterPositionAtTimeOfCast;
+        const velocity = math.similarTriangles(target.x - startPoint.x, target.y - casterPositionAtTimeOfCast.y, math.distance(startPoint, target), config.ARROW_PROJECTILE_SPEED)
+        let image: Image.IImageAnimated | undefined;
+        if (!prediction) {
+          image = Image.create(casterPositionAtTimeOfCast, 'projectile/arrow', containerProjectiles)
+          if (image) {
+            image.sprite.rotation = Math.atan2(velocity.y, velocity.x);
+          }
+        }
+        const pushedObject: HasSpace = {
+          x: casterPositionAtTimeOfCast.x,
+          y: casterPositionAtTimeOfCast.y,
+          radius: 1,
+          inLiquid: false,
+          image,
+          immovable: false,
+          beingPushed: false
+        }
+        makeForceMoveProjectile({
+          sourceUnit: state.casterUnit,
+          pushedObject,
+          startPoint,
+          velocity,
+          piercesRemaining: 0,
+          bouncesRemaining: quantity,
+          collidingUnitIds: [state.casterUnit.id],
+          collideFnKey: targetRicochetArrowCardId,
+          state,
+        }, underworld, prediction);
+
+        if (!prediction && !globalThis.headless) {
+          const timeout = Math.max(0, timeoutToNextArrow);
+          await new Promise(resolve => setTimeout(resolve, timeout));
+          // Decrease timeout with each subsequent arrow fired to ensure that players don't have to wait too long
+          timeoutToNextArrow -= 5;
+        }
+      }
+
+      await underworld.awaitForceMoves();
+      // - - - - - End copied from arrow.ts - - - - -
+
+      // Update targets variable to include newly added targets
+      targets = getCurrentTargets(state);
+      if (!prediction && !globalThis.headless && globalThis.predictionGraphics) {
+        const promises: Promise<void>[] = [];
+        targets.forEach(t => {
+          // Animations do not occur on headless
+          promises.push(new Promise<void>((resolve) => {
+            if (globalThis.predictionGraphics) {
+              globalThis.predictionGraphics.lineStyle(2, colors.targetingSpellGreen, 1.0)
+              playSFXKey('targetAquired');
```

Comment:
When animating, the target ricochet arrow doesn't show targets being acquired as it collides with them.  I think this would look/feel better

Reply:
I think maybe I just need to make this an automatic part of addTarget so we don't have to handle it individually in each targeting spell so actually leave that out of this for now and I'll address later